### ISSUE TITLE
wayland-egl.c is blocking

### ISSUE THREAD

**Issue by rtbo** [eyelash/tutorials#1]
Blocking in `wl_display_dispatch`, which should be replaced by `wl_display_dispatch_pending` as long as surface is not configured.
This is what `weston/clients/simple-egl.c` is doing.
Also what I am doing here in the D bindings: https://github.com/rtbo/wayland-d/blob/master/examples/egl_window/source/egl_window.d

**Comment by eyelash** [comment]
You mean calling `wl_display_dispatch` as long as the surface is not configured? Because that is what the [weston example](https://cgit.freedesktop.org/wayland/weston/tree/clients/simple-egl.c#n915) does. Your example does it the other way round.

**Comment by rtbo** [comment]
No, I'm doing it the same way, but my flag is reversed! (`wait_for_configure == !configured`)
AFAICT, calling `wl_display_dispatch` will block if the window is not configured. (I'm not pretending to understand why)
